src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java

```diff
@@ -0,0 +1,524 @@
+		List<Long> parentIds = StandardizationDryRunCompound.findParentIdsWithStandardizationChanges().getResultList();
+		logger.info("standardization initialized");
+		int result = restandardizeParentStructures(parentIds);
+		logger.info("standardization complete");
+		return(result);
+	}
+	
+	@Override
+	public String getDryRunStats() throws StandardizerException {
+		StandardizerSettingsConfigDTO standardizerConfigs = chemStructureService.getStandardizerSettings();
+		StandardizationHistory dryRunStats = new StandardizationDryRunCompound().fetchStats();
+		dryRunStats.setSettings(standardizerConfigs.toJson());
+		dryRunStats.setSettingsHash(standardizerConfigs.hashCode());
+		return dryRunStats.toJson();
+	}
+	
+	@Override
+	public StandardizationSettings getStandardizationSettings() {
+		List<StandardizationSettings> standardizationSettingses = StandardizationSettings.findAllStandardizationSettingses("modifiedDate", "DESC");
+		StandardizationSettings standardizationSettings = new StandardizationSettings();
+		if(standardizationSettingses.size() > 0) {
+			standardizationSettings = standardizationSettingses.get(0);
+		}
+		return(standardizationSettings);
+	}
+
+	@Override
+	public List<StandardizationHistory> getStanardizationHistory() {
+		List<StandardizationHistory> standardizationSettingses = StandardizationHistory.findStandardizationHistoryEntries(0, 500, "dateOfStandardization", "DESC");
+		return standardizationSettingses;
+	}
+
+	@Override
+	public void executeDryRun() throws CmpdRegMolFormatException, IOException, StandardizerException {
+		StandardizationHistory stndznHistory = getMostRecentStandardizationHistory();
+		StandardizerSettingsConfigDTO standardizationSettings = chemStructureService.getStandardizerSettings();
+		if(stndznHistory == null || StringUtils.equalsIgnoreCase(stndznHistory.getStandardizationStatus(), "complete")) {
+			stndznHistory = new StandardizationHistory();
+			stndznHistory.setSettings(standardizationSettings.toJson());
+			stndznHistory.setSettingsHash(standardizationSettings.hashCode());
+			stndznHistory.setRecordedDate(new Date());			
+		}
+		stndznHistory.setDryRunStatus("running");
+		stndznHistory.setDryRunStart(new Date());
+		stndznHistory.setDryRunComplete(null);
+		stndznHistory.persist();
+		
+	    int numberOfDisplayChanges = this.runDryRun();
+	    
+		stndznHistory.setDryRunComplete(new Date());
+		stndznHistory.setDryRunStatus("complete");
+		stndznHistory.persist();
+	}
+	
+	@Transactional
+	private int runDryRun() throws CmpdRegMolFormatException, IOException, StandardizerException {
+		logger.info("standardization dry run initialized");
+		logger.info("step 1/3: resetting dry run table");
+		this.reset();
+		logger.info("standardization dry run 1% complete");
+		logger.info("step 2/3: populating dry run table");
+		int numberOfDisplayChanges = this.populateStanardizationDryRunTable();  
+		logger.info("standardization dry run 20% complete");
```

Comment:
20% figure here is a bit strange. but I see the logging during populating the dry run table is separate. Did this come from an estimate that the restandardization takes 20% of the time and dupe checking takes 80% of the time?

There is already logging of % progress within step 2 and step 3. I think the "step 1/3, step 2/3" lines make sense, but the percentages don't make sense to me in this function. 1%, 20%, done?

Reply:
I removed these lines.  They were from some timings I did long ago when using jchem.